README.md

```diff
@@ -0,0 +1,315 @@
+# Terrajet - Generate Crossplane Providers from any Terraform Provider
+
+Terrajet is a code generator framework that allows developers to build code
+generation pipelines that can generate Crossplane controllers. Developers can
+start building their code generation pipeline targeting specific Terraform Providers
+by importing Terrajet and wiring all generators together, customizing the whole
+pipeline in the process.
+
+See [design document][design-doc] for more details.
+
+**NOTE**: Terrajet is in its very early stages. We expect many breaking changes
+in the coming weeks. Relying on it for production usage is not recommended yet.
+
+## Getting Started
+
+A Crossplane managed resource support consists of two main parts; type definition
+of `CustomResourceDefinition` that satisfy `xpresource.Managed` interface and
+an implementation of `managed.ExternalClient` acting on the instances of that
+`CustomResourceDefinition`.
+
+In Terrajet, we have code generators for generating the Go types of the resource
+from imported Terraform provider. Additionally, we have a common controller that
+implements `managed.ExternalClient`. Using `pipeline` package, you can generate
+the API types as well as the code that stitches it together with the implementation
+of `managed.ExternalClient`.
+
+The structure of the generator is just like any other Go program; we'll create
+a `cmd/generator/main.go` file and call Terrajet tools from there.
+
+The first step would be to use [`provider-template`][provider-template] to
```

Comment:
From my understanding, there are generally 3 components here:

1) A terraform provider e.g. https://github.com/rancher/terraform-provider-rancher2/ 
2) The Provider template https://github.com/crossplane/provider-template
and
3) Terrajet https://github.com/crossplane-contrib/terrajet

Possibly obvious to others but could you explain how these three fit together and where I would add the code generator?